pkg/controllers/awsloadbalancercontroller/deployment_test.go

```diff
@@ -0,0 +1,386 @@
+	}
+}
+
+func TestUpdateDeployment(t *testing.T) {
+	for _, tc := range []struct {
+		name               string
+		existingDeployment *appsv1.Deployment
+		desiredDeployment  *appsv1.Deployment
+		expectedDeployment *appsv1.Deployment
+	}{
+		{
+			name: "image changed",
+			existingDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).build(),
+			desiredDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v2"),
+			).build(),
+			expectedDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v2"),
+			).build(),
+		},
+		{
+			name: "replicas changed from value",
+			existingDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).withReplicas(1).build(),
+			desiredDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).withReplicas(2).build(),
+			expectedDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).withReplicas(2).build(),
+		},
+		{
+			name: "replicas changed from nil",
+			existingDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).build(),
+			desiredDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).withReplicas(1).build(),
+			expectedDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1"),
+			).withReplicas(1).build(),
+		},
+		{
+			name: "container args changed",
+			existingDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1", "--arg1", "--arg2"),
+			).build(),
+			desiredDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1", "--arg2", "--arg3"),
+			).build(),
+			expectedDeployment: testDeployment("operator", "test-namespace", "test-sa").withContainers(
+				testContainer("controller", "controller:v1", "--arg2", "--arg3"),
+			).build(),
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			client := fake.NewClientBuilder().WithObjects(tc.existingDeployment).Build()
+			r := &AWSLoadBalancerControllerReconciler{
+				Client: client,
+			}
+			currentDeployment, err := r.updateDeployment(context.Background(), tc.existingDeployment, tc.desiredDeployment)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+
+			if !equality.Semantic.DeepEqual(currentDeployment.Spec, tc.expectedDeployment.Spec) {
```

Comment:
What's the advantage of this comparison to `go-cmp`'s one?

Reply:
My assumption is that equality package is targeted at Kubernetes structs(no idea how) and `cmp.Diff` is more generic. That said I've switched over to `cmp.Diff` in the tests.